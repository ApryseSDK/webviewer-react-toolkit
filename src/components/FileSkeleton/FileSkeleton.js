'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.FileSkeleton = void 0;
var tslib_1 = require('tslib');
var classnames_1 = tslib_1.__importDefault(require('classnames'));
var react_1 = tslib_1.__importDefault(require('react'));
var FileSkeleton = function(_a) {
  var className = _a.className;
  var fileSkeletonClass = (0, classnames_1.default)(
    'ui__base ui__fileSkeleton',
    className,
  );
  return react_1.default.createElement(
    'div',
    { className: fileSkeletonClass },
    react_1.default.createElement('div', {
      className: 'ui__fileSkeleton__block ui__fileSkeleton__block--thumbnail',
    }),
    react_1.default.createElement('div', {
      className: 'ui__fileSkeleton__block ui__fileSkeleton__block--line-sm',
    }),
    react_1.default.createElement('div', {
      className: 'ui__fileSkeleton__block ui__fileSkeleton__block--line-xs',
    }),
    react_1.default.createElement('div', {
      className: 'ui__fileSkeleton__block ui__fileSkeleton__block--line-df',
    }),
    react_1.default.createElement('div', {
      className: 'ui__fileSkeleton__block ui__fileSkeleton__block--line-lgx',
    }),
    react_1.default.createElement('div', {
      className: 'ui__fileSkeleton__block ui__fileSkeleton__block--line-lg',
    }),
    react_1.default.createElement('div', {
      className: 'ui__fileSkeleton__block ui__fileSkeleton__block--line-df',
    }),
  );
};
exports.FileSkeleton = FileSkeleton;
//# sourceMappingURL=FileSkeleton.js.map
