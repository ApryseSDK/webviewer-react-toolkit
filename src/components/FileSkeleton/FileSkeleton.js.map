{"version":3,"file":"FileSkeleton.js","sourceRoot":"","sources":["FileSkeleton.tsx"],"names":[],"mappings":";;;;AAAA,kEAAoC;AACpC,wDAA0B;AASnB,IAAM,YAAY,GAAG,UAAC,EAAgC;QAA9B,SAAS,eAAA;IACtC,IAAM,iBAAiB,GAAG,IAAA,oBAAU,EAAC,2BAA2B,EAAE,SAAS,CAAC,CAAC;IAE7E,OAAO,CACL,uCAAK,SAAS,EAAE,iBAAiB;QAC/B,uCAAK,SAAS,EAAC,4DAA4D,GAAG;QAC9E,uCAAK,SAAS,EAAC,0DAA0D,GAAG;QAC5E,uCAAK,SAAS,EAAC,0DAA0D,GAAG;QAC5E,uCAAK,SAAS,EAAC,0DAA0D,GAAG;QAC5E,uCAAK,SAAS,EAAC,2DAA2D,GAAG;QAC7E,uCAAK,SAAS,EAAC,0DAA0D,GAAG;QAC5E,uCAAK,SAAS,EAAC,0DAA0D,GAAG,CACxE,CACP,CAAC;AACJ,CAAC,CAAC;AAdW,QAAA,YAAY,gBAcvB"}