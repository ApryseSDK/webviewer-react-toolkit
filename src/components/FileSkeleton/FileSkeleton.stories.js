'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.Basic = void 0;
var tslib_1 = require('tslib');
var react_1 = tslib_1.__importDefault(require('react'));
var FileSkeleton_1 = require('../FileSkeleton');
var README_md_1 = tslib_1.__importDefault(require('./README.md'));
exports.default = {
  title: 'Components/FileSkeleton',
  component: FileSkeleton_1.FileSkeleton,
  parameters: { readme: README_md_1.default },
};
var Basic = function() {
  return react_1.default.createElement(FileSkeleton_1.FileSkeleton, null);
};
exports.Basic = Basic;
//# sourceMappingURL=FileSkeleton.stories.js.map
