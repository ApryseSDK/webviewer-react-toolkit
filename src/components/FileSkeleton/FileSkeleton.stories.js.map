{"version":3,"file":"FileSkeleton.stories.js","sourceRoot":"","sources":["FileSkeleton.stories.tsx"],"names":[],"mappings":";;;;AAAA,wDAA0B;AAC1B,gDAA+C;AAC/C,kEAAiC;AAEjC,kBAAe,EAAE,KAAK,EAAE,yBAAyB,EAAE,SAAS,EAAE,2BAAY,EAAE,UAAU,EAAE,EAAE,MAAM,qBAAA,EAAE,EAAE,CAAC;AAE9F,IAAM,KAAK,GAAG,cAAM,OAAA,8BAAC,2BAAY,OAAG,EAAhB,CAAgB,CAAC;AAA/B,QAAA,KAAK,SAA0B"}