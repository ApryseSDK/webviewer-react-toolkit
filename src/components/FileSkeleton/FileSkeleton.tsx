import classnames from 'classnames';
import React, { FC } from 'react';

export interface FileSkeletonProps {
  /**
   * Classname of the skeleton wrapper
   */
  className?: string;
}

export const FileSkeleton: FC<FileSkeletonProps> = ({ className }) => {
  const fileSkeletonClass = classnames('ui__base ui__fileSkeleton', className);

  return (
    <div className={fileSkeletonClass}>
      <div className="ui__fileSkeleton__block ui__fileSkeleton__block--thumbnail" />
      <div className="ui__fileSkeleton__block ui__fileSkeleton__block--line-sm" />
      <div className="ui__fileSkeleton__block ui__fileSkeleton__block--line-xs" />
      <div className="ui__fileSkeleton__block ui__fileSkeleton__block--line-df" />
      <div className="ui__fileSkeleton__block ui__fileSkeleton__block--line-lgx" />
      <div className="ui__fileSkeleton__block ui__fileSkeleton__block--line-lg" />
      <div className="ui__fileSkeleton__block ui__fileSkeleton__block--line-df" />
    </div>
  );
};
