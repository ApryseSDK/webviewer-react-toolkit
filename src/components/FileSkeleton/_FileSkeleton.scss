.ui__fileSkeleton {
  background-color: $white-1;
  padding: 10px;
  height: 172px;
  width: 133px;

  &__block {
    @include skeleton;

    &--thumbnail {
      width: 40px;
      height: 40px;
      float: left;
    }

    &--line-sm {
      margin-left: 50px;
      margin-top: 10px;
      width: 60px;
      height: 6px;
    }

    &--line-xs {
      margin-left: 50px;
      margin-top: 10px;
      width: 60px;
      height: 6px;
    }

    &--line-df {
      margin-top: 20px;
      width: 95px;
      height: 6px;
    }

    &--line-lg {
      margin-top: 10px;
      width: 95px;
      height: 6px;
    }

    &--line-lgx {
      margin-top: 10px;
      width: 105px;
      height: 6px;
    }
  }
}
