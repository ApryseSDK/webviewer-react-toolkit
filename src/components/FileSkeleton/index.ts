export { FileSkeleton as default, FileSkeletonProps } from './FileSkeleton';
