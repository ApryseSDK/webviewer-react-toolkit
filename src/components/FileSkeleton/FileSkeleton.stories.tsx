import React from 'react';
import FileSkeleton from '../FileSkeleton';
import docs from './README.md';

export default { title: 'FileSkeleton', component: FileSkeleton, parameters: { info: docs } };

export const basic = () => <FileSkeleton />;
