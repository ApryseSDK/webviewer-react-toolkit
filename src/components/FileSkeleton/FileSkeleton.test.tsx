import { shallow } from 'enzyme';
import React from 'react';
import { FileSkeleton } from '../FileSkeleton';

describe('FileSkeleton component', () => {
  it('renders its contents', () => {
    const fileSkeleton = shallow(<FileSkeleton />);
    expect(fileSkeleton.find('.ui__fileSkeleton')).toHaveLength(1);
  });

  it('snapshot renders default fileSkeleton', () => {
    const fileSkeleton = shallow(<FileSkeleton />);
    expect(fileSkeleton).toMatchSnapshot();
  });
});
