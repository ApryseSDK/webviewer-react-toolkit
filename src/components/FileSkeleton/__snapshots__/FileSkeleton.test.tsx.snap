// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileSkeleton component snapshot renders default fileSkeleton 1`] = `
<div
  className="ui__base ui__fileSkeleton"
>
  <div
    className="ui__fileSkeleton__block ui__fileSkeleton__block--thumbnail"
  />
  <div
    className="ui__fileSkeleton__block ui__fileSkeleton__block--line-sm"
  />
  <div
    className="ui__fileSkeleton__block ui__fileSkeleton__block--line-xs"
  />
  <div
    className="ui__fileSkeleton__block ui__fileSkeleton__block--line-df"
  />
  <div
    className="ui__fileSkeleton__block ui__fileSkeleton__block--line-lgx"
  />
  <div
    className="ui__fileSkeleton__block ui__fileSkeleton__block--line-lg"
  />
  <div
    className="ui__fileSkeleton__block ui__fileSkeleton__block--line-df"
  />
</div>
`;
